gc.c

```diff
@@ -10914,15 +10919,17 @@ rb_gc_start(void)
 void
 rb_gc(void)
 {
-    rb_objspace_t *objspace = &rb_objspace;
+    rb_objspace_t *objspace = rb_objspace_ptr();
```

Comment:
Rather than a `rb_objspace_ptr` macro, wouldn't it be more convenient to implement some kind of `EARLY_RETURN_IF_SHUTTING_DOWN` kind of help macro?

Reply:
Just an idea 😜 

```C
#define unless_objspace(objspace) \
    rb_objspace_t *objspace; \
    rb_vm_t *rb_objspace_ptr_vm = GET_VM(); \
    if (rb_objspace_ptr_vm) objspace = rb_objspace_ptr_vm->objspace; \
    else /* return; or objspace will be warned uninitialized */

unless_objspace(objspace) {return;}
// or
unless_objspace(objspace) {return FALSE;}
```

Reply:
I tried out the early return approach, but doing it right involved some `do while (0)` trickery, and I couldn’t quite get it to properly return default values.

This approach seemed simpler to implement and easier to understand, but happy to try another approach. `unless_objspace` seems good, I’ll give it a whirl.

Reply:
Implemented in https://github.com/ruby/ruby/pull/7783/commits/77ad19c453d1165407608c82163dda5122ccdce7